File: 8kyu/(8 kyu) String cleaning/(8 kyu) String cleaning.java
// 1 Plain solution
class StringCleaning {
    static String stringClean(final String text) {
        var res = new StringBuilder();
        for (char c : text.toCharArray()) {
            if (Character.isDigit(c)) {
                // If the character is a digit, skip it
                continue;
            }
            res.append(c);
        }
        return res.toString();
    }
}

// 2 Optimized solution
import java.util.stream.Collectors;

class StringCleaning {
    static String stringClean(final String text) {
        return text.chars()
                   .filter(c -> !Character.isDigit(c))
                   .mapToObj(c -> (char) c)
                   .map(String::valueOf)
                   .collect(Collectors.joining());
    }
}

// 3 Clever solution
class StringCleaning {
    static String stringClean(final String text) {
        return text.replaceAll("\\d", "");
    }
}

File: 8kyu/(8 kyu) String cleaning/(8 kyu) String cleaning.cs
// 1 Simple solution
using System;
using System.Text;

public class Kata
{
  public static string StringClean(string s)
  {
    var res = new StringBuilder();
    for (int i = 0; i < s.Length; i++)
    {
      if (!char.IsDigit(s[i]))
      {
        res.Append(s[i]);
      }
    }
    return res.ToString();
  }
}

// 2 Optimized solution
using System;
using System.Linq;

public class Kata
{
    public static string StringClean(string s)
    {
        return new string(s.Where(c => !char.IsDigit(c)).ToArray());
    }
}

// 3 Clever solution
public class Kata
{
    public static string StringClean(string s)
    {
        return System.Text.RegularExpressions.Regex.Replace(s, "[0-9]", "");
    }
}
